Sources/SwiftDocC/Infrastructure/NodeURLGenerator.swift

```diff
@@ -194,7 +194,13 @@ public struct NodeURLGenerator {
                     isURLModified = true
                     name = "'\(name)"
                 }
-                
+
+                let components = name.components(separatedBy: ["<", ">", ":", "\"", "/", "\\", "|", "?", "*"])
+                if components.count > 1 {
+                    isURLModified = true
+                    name = components.joined(separator: "_")
```

Comment:
I wonder if we should always append a hashed identifier to the name here to ensure that the file names don't collide.

Reply:
I think that appending the hashed identifier seems reasonable.  Definitely would avoid any possible conflicts.  Would we do that for any case where `isURLModified` is true?

Reply:
The code already _mostly_ does that when modifying the URL, although if it's a specific path component that's longer than what a file name is allowed to be, then it shortens that path component and appends the hash to that component instead of later in the URL.